utam-compiler/src/main/java/utam/compiler/translator/JsonCompilerConfig.java

```diff
@@ -0,0 +1,364 @@
+      return all;
+    }
+
+    /**
+     * get list of profiles from config
+     *
+     * @return list
+     */
+    public List<Profile> getRawProfiles() {
+      return profiles;
+    }
+
+    /**
+     * file mask to recognize JSON with a page object
+     *
+     * @return string with mask
+     */
+    public String getPageObjectFileMask() {
+      return pageObjectFileMask;
+    }
+  }
+
+  /**
+   * Represents a profile configuration for a UTAM Page Object repository.
+   *
+   * @author james.evans
+   */
+  public static class Profile {
+
+    static final String ERR_DUPLICATE_PROFILE_DIFF_VALUES = "Profile '%s' is already configured with different values";
+    private final String name;
+    private final String[] values;
+
+    /**
+     * Initializes a new instance of the Profile class. Instantiated via JSON deserialization.
+     *
+     * @param name   the name of the profile
+     * @param values the values of the profile
+     */
+    @JsonCreator
+    public Profile(
+        @JsonProperty(value = "name") String name,
+        @JsonProperty(value = "values") String[] values) {
+      this.name = name;
+      this.values = values;
+    }
+
+    public ProfileConfiguration getProfileConfiguration() {
+      return new StringValueProfileConfig(name, values);
+    }
+
+    public void setConfiguredProfile(List<ProfileConfiguration> profiles) {
+      ProfileConfiguration profileConfiguration = getProfileConfiguration();
+      for (ProfileConfiguration alreadyConfigured : profiles) {
+        if (profileConfiguration.equals(alreadyConfigured)) {
+          // same profile config already added
+          return;
+        }
+        // check for profile with same name but different values
+        if (alreadyConfigured.getPropertyKey().equals(profileConfiguration.getPropertyKey())) {
+          throw new UtamCompilationError(
+              String.format(ERR_DUPLICATE_PROFILE_DIFF_VALUES, alreadyConfigured.getPropertyKey()));
+        }
+      }
+      profiles.add(profileConfiguration);
+    }
+  }
+
+  /**
+   * Represents a namespace within a UTAM Page Object library.
+   *
+   * @author james.evans
+   */
+  public static class Namespace {
+
+    static final String ERR_DUPLICATE_MAPPING = "Namespace config: value for '%s' is already configured as '%s'";
+
+    private final String typePrefix;
+    private final String pathMatchRegex;
+
+    /**
+     * Initializes a new instance of the Namespace class. Instantiated via JSON deserialization.
+     *
+     * @param typePrefix     type prefix
+     * @param pathMatchRegex the mapped name used by the UTAM compiler to generate packages for the
+     *                       generated Page Object source code
+     */
+    @JsonCreator
+    public Namespace(
+        @JsonProperty(value = "typePrefix", required = true) String typePrefix,
+        @JsonProperty(value = "pathMatchRegex", required = true) String pathMatchRegex) {
```

Comment:
@jimevanssfdc let me know if you're ok with property names for namespace

Reply:
I'm entirely agnostic on property names. These seem perfectly fine to me. They're descriptive and their purpose is clear.